File: Pascal/ЕГЭ2021;2022/23/Решение заданий. Поляков/3710.pas
uses school;
function f(start, endn:integer):integer;
begin
  if start = endn then f:=1
  else if start > endn then f:=0
  else f:=f(start + 1, endn)+f(start * 2, endn)+
                              f(start * 2 +1, endn)
end;

begin
  Writeln('Ответ: ', f(5, Dec('101110', 2)))
end.

File: Python/ЕГЭ2021;2022/23/Решение заданий. Поляков/3710.py
def f(start, end):
    if start == end:
        return 1
    elif int(start, 2) > int(end, 2):
        return 0
    else:
        rez = int(start, 2) + 1
        return f(bin(rez)[2:], end)+f(start+'0', end)+f(start+'1', end)


print(f('101', '101110'))
